Add sysparm_limit option to now.yml

ecchong opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Added option to allow changing sysparm_limit for query
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
The default ServiceNow REST Table API request timeout is 60 seconds.  Sometime the API call is not able to complete in 60 seconds and timeout, which causing the whole inventory sync job to fail.  This option allow user to change the hard-coded 1000 batch size to a small number in case they are experiencing timeout issue.  If the option is not provided, it defaults back to 1000.

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
servicenow.itsm.now

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
Example of `now.yml` with new `sysparm_limit` option
```paste below
---
plugin: servicenow.itsm.now
table: cmdb_ci_server
strict: false

columns:
  - sys_id
  - name
  - fqdn
  - serial_number
  - ip_address
  - os
  - os_version
  - asset.install_status
  - support_group.name
  - location.name
  
sysparm_query: os=Linux Red Hat
sysparm_limit: 500
... 
```

tupyy commented on changelogs/changelog.yaml:
```diff
@@ -305,3 +305,9 @@ releases:
     - client_validate_certs.yaml
     - release_2_3_0.yml
     release_date: '2023-07-25'
+  2.4.0:
```
changelog is automatically generated. No need to add anything here

tupyy commented on plugins/inventory/now.py:
```diff
@@ -143,6 +143,13 @@
+      - Changing the query batch size
+      - If not set, default to 1000
+    type: int
+    version_added: 2.3.0
```
I think it's "2.4.0"

tupyy commented:
@ecchong Thank you for your contributions.
except some doc stuff /lgtm

tupyy commented:
@ecchong You need to provide the fragment file. The rest is generated.

ecchong commented:
> @ecchong You need to provide the fragment file. The rest is generated.

OK.  Reverted changelogs/changelog.yaml back.  Just updated changelogs/fragments/sysparm_limit.yml.
Will docs/servicenow.itsm.now_inventory.rst update automaticatlly?

tupyy commented:
Yes

tupyy commented:
/lgtm

tupyy commented:
@ecchong  Thank you

Akasurde commented on plugins/inventory/now.py:
```diff
@@ -516,7 +523,11 @@ def parse(self, inventory, loader, path, cache=True):
-        table_client = TableClient(client)
+        sysparm_limit = self.get_option("sysparm_limit")
+        if sysparm_limit:
+            table_client = TableClient(client, sysparm_limit)
```
```suggestion
            table_client = TableClient(client, batch_size=sysparm_limit)
```

Akasurde commented:
@tupyy Could you please take a look?

tupyy commented:
> @tupyy Could you please take a look?

Yes. I'll look.

tupyy commented on tests/integration/targets/inventory/playbooks/sysparm_limit.yml:
```diff
@@ -0,0 +1,38 @@
+  gather_facts: false
+
+  tasks:
+    - block:
```
Please use `sysparm_limit` in your test. This is the functionality we want to test.

tupyy commented:
@ecchong  nice work. thank you
